dast/diff.py

```diff
@@ -0,0 +1,336 @@
+"""
+This script contains only modifications to deepdiff core classes.
+The edits are clearly highlighted through comments, anything else is just copy pasted from source
+"""
+
+
+from deepdiff.diff import (PASSES_COUNT, notpresent_indexed, SubscriptableIterableRelationship,
+                           DeepDiff, OrderedSetPlus, notpresent, add_to_frozen_set)
+
+from deepdiff.deephash import (DeepHash, EMPTY_FROZENSET, dict_, logger, get_id, KEY_TO_VAL_STR, unprocessed)
+
+class Hash(DeepHash):
+    def _prep_iterable(self, obj, parent, parents_ids=EMPTY_FROZENSET):
+
+        counts = 1
+# --- begin diff wrt to deepdiff ---
+        result = set()
+# --- end diff wrt to deepdiff ---
+
+        for i, item in enumerate(obj):
+            new_parent = "{}[{}]".format(parent, i)
+            if self._skip_this(item, parent=new_parent):
+                continue
+
+            item_id = get_id(item)
+            if parents_ids and item_id in parents_ids:
+                continue
+
+            parents_ids_added = add_to_frozen_set(parents_ids, item_id)
+            hashed, count = self._hash(item, parent=new_parent, parents_ids=parents_ids_added)
+            # counting repetitions
+            counts += count
+# --- begin diff wrt to deepdiff ---
+            result.add((hashed, i))
+
+        if self.ignore_repetition:
+            result = list(set(map(lambda x: x[0], result)))
+        else:
+            result = [
+                f'{hashed}|{i}' for hashed, i in result
+            ]
+# --- end diff wrt to deepdiff ---
+
+        result = sorted(map(str, result))  # making sure the result items are string and sorted so join command works.
+        result = ','.join(result)
+        result = KEY_TO_VAL_STR.format(type(obj).__name__, result)
+
+        return result, counts
+
+class Diff(DeepDiff):
+    mapping = {}
+
+    def _get_matching_pairs(self, level):
```

Comment:
This function isn't called when you do `ignore_order`, is it?

Reply:
Ah yes thanks, I was playing directly in DeepDiff and when I moved the logic here I may have added unnecessary changes, I will go through and remove those